shop/forms/checkout.py

```diff
@@ -69,7 +69,7 @@ def __init__(self, initial=None, instance=None, *args, **kwargs):
     def form_factory(cls, request, data, cart):
         customer_form = cls(data=data, instance=request.customer.user)
         if customer_form.is_valid():
-            customer_form.instance.recognize_as_guest(request, commit=False)
+            customer_form.instance.customer.recognize_as_guest(request, commit=False)
```

Comment:
wouldn't it be better to just invoke ``request.customer.recognize_as_guest(request, commit=False)``
..., well this was already wrong in the first place.